### ISSUE TITLE
Breaking change in WTForms 3.2.0 leads to internal error

### ISSUE THREAD

**Issue by ClementDelannoySlateDigital** [pallets-eco/flask-admin#2572]
The callstack: 

api-1  |   File "/usr/local/lib/python3.12/site-packages/flask/app.py", line 1536, in __call__
api-1  |     return self.wsgi_app(environ, start_response)
api-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
api-1  |   File "/usr/local/lib/python3.12/site-packages/flask/app.py", line 1514, in wsgi_app
api-1  |     response = self.handle_exception(e)
api-1  |                ^^^^^^^^^^^^^^^^^^^^^^^^
api-1  |   File "/usr/local/lib/python3.12/site-packages/flask/app.py", line 1511, in wsgi_app
api-1  |     response = self.full_dispatch_request()
api-1  |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
api-1  |   File "/usr/local/lib/python3.12/site-packages/flask/app.py", line 919, in full_dispatch_request
api-1  |     rv = self.handle_user_exception(e)
api-1  |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
api-1  |   File "/usr/local/lib/python3.12/site-packages/flask/app.py", line 917, in full_dispatch_request
api-1  |     rv = self.dispatch_request()
api-1  |          ^^^^^^^^^^^^^^^^^^^^^^^
api-1  |   File "/usr/local/lib/python3.12/site-packages/flask/app.py", line 902, in dispatch_request
api-1  |     return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
api-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
api-1  |   File "/usr/local/lib/python3.12/site-packages/flask_admin/base.py", line 69, in inner
api-1  |     return self._run_view(f, *args, **kwargs)
api-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
api-1  |   File "/usr/local/lib/python3.12/site-packages/flask_admin/base.py", line 369, in _run_view
api-1  |     return fn(self, *args, **kwargs)
api-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^
api-1  |   File "/usr/local/lib/python3.12/site-packages/flask_admin/model/base.py", line 2149, in edit_view
api-1  |     form = self.edit_form(obj=model)
api-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^
api-1  |   File "/usr/local/lib/python3.12/site-packages/flask_admin/model/base.py", line 1340, in edit_form
api-1  |     return self._edit_form_class(get_form_data(), obj=obj)
api-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
api-1  |   File "/usr/local/lib/python3.12/site-packages/wtforms/form.py", line 209, in __call__
api-1  |     return type.__call__(cls, *args, **kwargs)
api-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
api-1  |   File "/usr/local/lib/python3.12/site-packages/flask_admin/form/__init__.py", line 22, in __init__
api-1  |     super(BaseForm, self).__init__(formdata=formdata, obj=obj, prefix=prefix, **kwargs)
api-1  |   File "/usr/local/lib/python3.12/site-packages/wtforms/form.py", line 281, in __init__
api-1  |     super().__init__(self._unbound_fields, meta=meta_obj, prefix=prefix)
api-1  |   File "/usr/local/lib/python3.12/site-packages/wtforms/form.py", line 49, in __init__
api-1  |     field = meta.bind_field(self, unbound_field, options)
api-1  |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
api-1  |   File "/usr/local/lib/python3.12/site-packages/wtforms/meta.py", line 28, in bind_field
api-1  |     return unbound_field.bind(form=form, **options)
api-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
api-1  |   File "/usr/local/lib/python3.12/site-packages/wtforms/fields/core.py", line 387, in bind
api-1  |     return self.field_class(*self.args, **kw)
api-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
api-1  |   File "/usr/local/lib/python3.12/site-packages/wtforms/fields/core.py", line 133, in __init__
api-1  |     for k, v in flags.items():
api-1  |                 ^^^^^^^^^^^^^^

Environment:

docker python:3.12

- Python version: python 3.12
- Flask version: 3.1.0
- Flask-Admin version: 1.6.1

**Comment by HK-Mattew** [comment]
I have this same problem. And downgrading the WTForms lib to 3.1.2 the problem does not occur.

**Comment by samuelhwilliams** [comment]
Would you mind testing this with version v2.0.0a2 and let me know if this problem is present there, too?

**Comment by ClementDelannoySlateDigital** [comment]
Hi Samuel, thank you for your response

Trying with Flask-Admin 2.0.0a2, WTForms 3.2.1, Flask 3.1.0:

```
api-1  |   File "/usr/local/lib/python3.12/site-packages/flask_admin/_compat.py", line 49, in _iter_choices_wtforms_compat
api-1  |     from packaging.version import Version
api-1  | ModuleNotFoundError: No module named 'packaging'
```

Seems like an undeclared dependency. My scenario work fine after installing packaging.

**Comment by samuelhwilliams** [comment]
This should be fixed in the next release (v2.0.0a3)
